src/StringState.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * This file is part of the php-state project.
+ *
+ * (c) Yannick Voyer <star.yvoyer@gmail.com> (http://github.com/yvoyer)
+ */
+
+namespace Star\Component\State;
+
+use Star\Component\State\Attribute\StateAttribute;
+use Webmozart\Assert\Assert;
+
+final class StringState implements State
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var StateAttribute[]
+     */
+    private $attributes = [];
+
+    public function __construct($id)
```

Comment:
Add attributes in construct. Make it behave as value object when adding attributes